vb-fe/src/components/player/PlayerWrapper.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FunctionComponent, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { usePlayerDevice, useErrorState, usePlaybackState } from "core/player/index";
+import { PlayerConstants } from "_store/player/PlayerTypes";
+import { SystemConstants } from "_store/system/SystemTypes";
+import { resetApp } from "util/Logout";
+
+export const PlayerWrapper: FunctionComponent = () => {
+  const device = usePlayerDevice();
+  const error = useErrorState();
+  const playbackState = usePlaybackState();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch({ type: SystemConstants.LOADING, payload: "Initializing Player..." });
+    if (device) {
+      dispatch({ type: PlayerConstants.SET_DEVICE_ID, payload: device.device_id });
+      dispatch({ type: SystemConstants.SUCCESS });
+    }
+  }, [device, dispatch]);
+
+  useEffect(() => {
+    if (error) {
+      switch (error.type) {
+        case "authentication_error":
+          resetApp();
+          dispatch({ type: SystemConstants.LOGIN, payload: false });
+          break;
+        default:
+          resetApp();
+          dispatch({ type: SystemConstants.LOGIN, payload: false });
```

Comment:
These are both doing the same thing, why the `switch`? Are we anticipating other error types?

Reply:
ohh yeah there are a few more error types, i just have written them down